### PR TITLE
XW-3476: Fix usage of global state in Wikia Rail

### PR DESCRIPTION
**Pull request by mszabo-wikia**
@Wikia/x-wing

### CODE REVIEW

File: skins/oasis/modules/BodyController.class.php [hunk 1]
```diff
@@ -206,7 +209,7 @@ public function getRailModuleList() {
 		}
 
 		// Content, category and forum namespaces.  FB:1280 Added file,video,mw,template
-		if ( $this->wg->Title->isSubpage() && $this->wg->Title->getNamespace() == NS_USER ||
+		if ( $title->isSubpage() && $title->getNamespace() == NS_USER ||
```
Comment (WikiaTech) [inline skins/oasis/modules/BodyController.class.php:212]: MAJOR Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)

File: skins/oasis/modules/BodyController.class.php [hunk 2]
```diff
@@ -141,13 +141,16 @@ public static function getUserPagesNamespaces() {
 	}
 
 	public function getRailModuleList() {
```
Comment (WikiaTech) [inline skins/oasis/modules/BodyController.class.php:143]: MAJOR The Cyclomatic Complexity of this function "getRailModuleList" is 40 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
MAJOR Reduce the number of returns of this function 6, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 64 issues
* ![MAJOR][MAJOR] 59 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WikiaDispatchableObject.class.php#L12](https://github.com/Wikia/app/blob/50d7a6dff9d348b012edf721d5f3ae3ba8f2df7f/includes/wikia/nirvana/WikiaDispatchableObject.class.php#L12): Class "WikiaDispatchableObject" has 33 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [WikiaDispatchableObject.class.php#L213](https://github.com/Wikia/app/blob/50d7a6dff9d348b012edf721d5f3ae3ba8f2df7f/includes/wikia/nirvana/WikiaDispatchableObject.class.php#L213): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L29](https://github.com/Wikia/app/blob/50d7a6dff9d348b012edf721d5f3ae3ba8f2df7f/skins/oasis/modules/BodyController.class.php#L29): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L30](https://github.com/Wikia/app/blob/50d7a6dff9d348b012edf721d5f3ae3ba8f2df7f/skins/oasis/modules/BodyController.class.php#L30): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L38](https://github.com/Wikia/app/blob/50d7a6dff9d348b012edf721d5f3ae3ba8f2df7f/skins/oasis/modules/BodyController.class.php#L38): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L38](https://github.com/Wikia/app/blob/50d7a6dff9d348b012edf721d5f3ae3ba8f2df7f/skins/oasis/modules/BodyController.class.php#L38): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L45](https://github.com/Wikia/app/blob/50d7a6dff9d348b012edf721d5f3ae3ba8f2df7f/skins/oasis/modules/BodyController.class.php#L45): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L51](https://github.com/Wikia/app/blob/50d7a6dff9d348b012edf721d5f3ae3ba8f2df7f/skins/oasis/modules/BodyController.class.php#L51): Reduce the number of returns of this function 6, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR][MAJOR] [BodyController.class.php#L67](https://github.com/Wikia/app/blob/50d7a6dff9d348b012edf721d5f3ae3ba8f2df7f/skins/oasis/modules/BodyController.class.php#L67): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L86](https://github.com/Wikia/app/blob/50d7a6dff9d348b012edf721d5f3ae3ba8f2df7f/skins/oasis/modules/BodyController.class.php#L86): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by rogatty — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 64 issues
* ![MAJOR][MAJOR] 59 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WikiaDispatchableObject.class.php#L12](https://github.com/Wikia/app/blob/1b1e4a38530c1556177f2060631bc468ea519b05/includes/wikia/nirvana/WikiaDispatchableObject.class.php#L12): Class "WikiaDispatchableObject" has 33 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [WikiaDispatchableObject.class.php#L213](https://github.com/Wikia/app/blob/1b1e4a38530c1556177f2060631bc468ea519b05/includes/wikia/nirvana/WikiaDispatchableObject.class.php#L213): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L29](https://github.com/Wikia/app/blob/1b1e4a38530c1556177f2060631bc468ea519b05/skins/oasis/modules/BodyController.class.php#L29): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L30](https://github.com/Wikia/app/blob/1b1e4a38530c1556177f2060631bc468ea519b05/skins/oasis/modules/BodyController.class.php#L30): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L38](https://github.com/Wikia/app/blob/1b1e4a38530c1556177f2060631bc468ea519b05/skins/oasis/modules/BodyController.class.php#L38): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L38](https://github.com/Wikia/app/blob/1b1e4a38530c1556177f2060631bc468ea519b05/skins/oasis/modules/BodyController.class.php#L38): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L45](https://github.com/Wikia/app/blob/1b1e4a38530c1556177f2060631bc468ea519b05/skins/oasis/modules/BodyController.class.php#L45): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L51](https://github.com/Wikia/app/blob/1b1e4a38530c1556177f2060631bc468ea519b05/skins/oasis/modules/BodyController.class.php#L51): Reduce the number of returns of this function 6, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR][MAJOR] [BodyController.class.php#L67](https://github.com/Wikia/app/blob/1b1e4a38530c1556177f2060631bc468ea519b05/skins/oasis/modules/BodyController.class.php#L67): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L86](https://github.com/Wikia/app/blob/1b1e4a38530c1556177f2060631bc468ea519b05/skins/oasis/modules/BodyController.class.php#L86): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 64 issues
* ![MAJOR][MAJOR] 59 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WikiaDispatchableObject.class.php#L12](https://github.com/Wikia/app/blob/5b303f21a8830927b397ec2ef799e31db83901fa/includes/wikia/nirvana/WikiaDispatchableObject.class.php#L12): Class "WikiaDispatchableObject" has 33 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [WikiaDispatchableObject.class.php#L213](https://github.com/Wikia/app/blob/5b303f21a8830927b397ec2ef799e31db83901fa/includes/wikia/nirvana/WikiaDispatchableObject.class.php#L213): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L29](https://github.com/Wikia/app/blob/5b303f21a8830927b397ec2ef799e31db83901fa/skins/oasis/modules/BodyController.class.php#L29): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L30](https://github.com/Wikia/app/blob/5b303f21a8830927b397ec2ef799e31db83901fa/skins/oasis/modules/BodyController.class.php#L30): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L38](https://github.com/Wikia/app/blob/5b303f21a8830927b397ec2ef799e31db83901fa/skins/oasis/modules/BodyController.class.php#L38): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L38](https://github.com/Wikia/app/blob/5b303f21a8830927b397ec2ef799e31db83901fa/skins/oasis/modules/BodyController.class.php#L38): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L45](https://github.com/Wikia/app/blob/5b303f21a8830927b397ec2ef799e31db83901fa/skins/oasis/modules/BodyController.class.php#L45): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L51](https://github.com/Wikia/app/blob/5b303f21a8830927b397ec2ef799e31db83901fa/skins/oasis/modules/BodyController.class.php#L51): Reduce the number of returns of this function 6, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR][MAJOR] [BodyController.class.php#L67](https://github.com/Wikia/app/blob/5b303f21a8830927b397ec2ef799e31db83901fa/skins/oasis/modules/BodyController.class.php#L67): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L86](https://github.com/Wikia/app/blob/5b303f21a8830927b397ec2ef799e31db83901fa/skins/oasis/modules/BodyController.class.php#L86): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 73 issues
* ![MAJOR][MAJOR] 68 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [EditPageLayoutHelper.class.php#L26](https://github.com/Wikia/app/blob/94f91b089c59161503ea6af6cde974ed9c795d5c/extensions/wikia/EditPageLayout/EditPageLayoutHelper.class.php#L26): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [EditPageLayoutHelper.class.php#L27](https://github.com/Wikia/app/blob/94f91b089c59161503ea6af6cde974ed9c795d5c/extensions/wikia/EditPageLayout/EditPageLayoutHelper.class.php#L27): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [EditPageLayoutHelper.class.php#L57](https://github.com/Wikia/app/blob/94f91b089c59161503ea6af6cde974ed9c795d5c/extensions/wikia/EditPageLayout/EditPageLayoutHelper.class.php#L57): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [EditPageLayoutHelper.class.php#L144](https://github.com/Wikia/app/blob/94f91b089c59161503ea6af6cde974ed9c795d5c/extensions/wikia/EditPageLayout/EditPageLayoutHelper.class.php#L144): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [EditPageLayoutHelper.class.php#L178](https://github.com/Wikia/app/blob/94f91b089c59161503ea6af6cde974ed9c795d5c/extensions/wikia/EditPageLayout/EditPageLayoutHelper.class.php#L178): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR][MAJOR] [EditPageLayoutHelper.class.php#L204](https://github.com/Wikia/app/blob/94f91b089c59161503ea6af6cde974ed9c795d5c/extensions/wikia/EditPageLayout/EditPageLayoutHelper.class.php#L204): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [EditPageLayoutHelper.class.php#L225](https://github.com/Wikia/app/blob/94f91b089c59161503ea6af6cde974ed9c795d5c/extensions/wikia/EditPageLayout/EditPageLayoutHelper.class.php#L225): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [EditPageLayoutHelper.class.php#L296](https://github.com/Wikia/app/blob/94f91b089c59161503ea6af6cde974ed9c795d5c/extensions/wikia/EditPageLayout/EditPageLayoutHelper.class.php#L296): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS112)
1. ![MAJOR][MAJOR] [EditPageLayoutHelper.class.php#L303](https://github.com/Wikia/app/blob/94f91b089c59161503ea6af6cde974ed9c795d5c/extensions/wikia/EditPageLayout/EditPageLayoutHelper.class.php#L303): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS112)
1. ![MAJOR][MAJOR] [WikiaDispatchableObject.class.php#L12](https://github.com/Wikia/app/blob/94f91b089c59161503ea6af6cde974ed9c795d5c/includes/wikia/nirvana/WikiaDispatchableObject.class.php#L12): Class "WikiaDispatchableObject" has 33 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**